pkg/controllers/webhookauthenticator/webhookauthenticator_controller.go

```diff
@@ -176,3 +164,61 @@ func (c *webhookAuthenticatorController) ensureKubeConfigSecret(ctx context.Cont
 	secret, _, err := resourceapply.ApplySecret(c.secrets, recorder, requiredSecret)
 	return secret, err
 }
+
+func (c *webhookAuthenticatorController) getAuthenticatorCertKeyPair() (key, cert []byte, err error) {
+	// waitingForCertKeyMsg lets us decide whether we should set progressing condition to true
+	// 1. nil: keep "progressing the same"
+	// 2. non-empty string: progress with the message in the string
+	// 3. empty string: stop progressing
+	var waitingForCertKeyMsg *string
+	defer func() {
+		if waitingForCertKeyMsg != nil {
+			cond := operatorv1.OperatorCondition{
+				Type:    "AuthenticatorCertKeyProgressing",
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "AsExpected",
+				Message: "All is well",
+			}
+
+			if len(*waitingForCertKeyMsg) > 0 {
+				cond.Status = operatorv1.ConditionTrue
+				cond.Reason = "WaitingForCertKey"
+				cond.Message = *waitingForCertKeyMsg
+			}
+
+			if _, _, statusErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(cond)); statusErr != nil {
+				klog.Errorf("failed to update operator status: %v", statusErr)
+				err = statusErr
```

Comment:
i hate named return values :-)

Reply:
I'm afraid there's no other way to propagate the status err to the return value :-|